app/advancedfilters/filterService.js

```diff
@@ -63,17 +63,18 @@ app.service('filterService', function(){
     });
   };
 
-  this.mergeRightToLeft = function(array1, array2) {
-    var concat = array1; //copy the array
-    var map1 = _.indexBy(array1,'id');
-    angular.forEach(array2, function(item) {
-      if(map1[item.id]) {
-        map1[item.id].item = item.item;
+  this.mergeArrays = function(dst, src) {
+    var concat = dst.slice(); //copy the array
+    var dstMap = _.indexBy(dst,'id');
+    angular.forEach(src, function(item) {
+      if(dstMap[item.id]) {
+        dstMap[item.id].checked = item.checked;
+        dstMap[item.id].item = (item.item) ? item.item : dstMap[item.id].item;
       } else {
         concat.push(item);
       }
     });
-    return concat;
+    return _.sortBy(concat, 'checked').reverse();
```

Comment:
Some of the checked items from `src` might already be in `dst`, so we need to sort at by checked before returning the concatenated array. Underscore only provides one direction for sorting, so the `reverse()` is required to provide the correct order